api/v1beta1/ccruntime_types.go

```diff
@@ -168,6 +168,16 @@ type CcInstallConfig struct {
 	// +optional
 	DefaultRuntimeClassName string `json:"defaultRuntimeClassName,omitempty"`
 
+	// This specifies the HTTPS_PROXY that may be required to be set for the Kata Containers agent
+	// when performing the image pull inside the guest (either using nydus snapshotter with containerd
+	// or CRI-O)
+	AgentHttpsProxy string `json:"agentHttpsProxy,omitempty"`
+
+	// This specifies the NO_PROXY that may be required to be set for the Kata Containers agent
+	// when performing the image pull inside the guest (either using nydus snapshotter with containerd
+	// or CRI-O)
+	AgentNoProxy string `json:"agentNoProxy,omitempty"`
+
```

Comment:
Since this is kata runtime specific (and I'm assuming doesn't apply to enclave-cc), how about putting these in a separate Kata agent specific struct so that the code is easier to read and update if new agent config params gets added ?

Reply:
That's a very good point, @bpradipt, let me revisit this one soon.

Reply:
Is the same not possible with `EnvironmentVariables`? Do you have the code somewhere that takes these into the kata agent?

Reply:
Based on ^^^ I added a do-not-merge label.